functions/firebase/FirestoreReactive/Function.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_firestore_reactive]
+using CloudNative.CloudEvents;
+using Google.Cloud.Firestore;
+using Google.Cloud.Functions.Framework;
+using Google.Cloud.Functions.Invoker;
+using Google.Events.Protobuf.Cloud.Firestore.V1;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+// Register the startup class to provide the Firestore dependency.
+[assembly: FunctionsStartup(typeof(FirestoreReactive.Startup))]
+
+namespace FirestoreReactive
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void ConfigureServices(WebHostBuilderContext context, IServiceCollection services) =>
+            services.AddSingleton(FirestoreDb.Create());
+    }
+
+    public class Function : ICloudEventFunction<DocumentEventData>
```

Comment:
I think this one doesn't have a test.

Reply:
You're right. The tricky part is that we'd need to fake out FirestoreClient. I can potentially do that... not sure whether it's worth it though. Thoughts welcome :)